pageserver/src/virtual_file.rs

```diff
@@ -582,24 +582,37 @@ impl VirtualFile {
     }
 
     // Copied from https://doc.rust-lang.org/1.72.0/src/std/os/unix/fs.rs.html#219-235
-    pub async fn write_all_at(&self, mut buf: &[u8], mut offset: u64) -> Result<(), Error> {
+    pub async fn write_all_at<B: BoundedBuf>(
+        &self,
+        buf: B,
+        mut offset: u64,
+    ) -> (B::Buf, Result<(), Error>) {
```

Comment:
nit: could return `Result<B::Buf, Error>` to force callers to inspect the error.

Reply:
The `Result` is `#[must_use]`, so, rustc should still warn if you don't handle the error.
I don't want to change the convention I stuck to throughout this whole patch series: you put in resources and you always get them back, along with a result.

Reply:
Ohhhh, actually `#[must_use]` doesn't work on tuples 🧠  https://github.com/rust-lang/rust/issues/39524

Reply:
I'll create a follow-up issue that I'll tend to after merging the patch series.

Reply:
https://github.com/neondatabase/neon/issues/6754